### ISSUE TITLE
Add buttons with shadows like in the ESC menu

### ISSUE THREAD

**Issue by necropotame** [teeworlds-modapi/teeworlds#60]
When the player press ESC, a list of buttons is display in the top and the bottom of the screen.
The server should be able to add new buttons in the left or right part, like "options", "help", "leadboard", ... These buttons could be displayers with a particular color to show clearly that these buttons are mod-related.

@Henningstone is automatically assigned to this task :-p What do you think about this idea ?

**Comment by ghost** [comment]
So.... Basically extended version of netgui cause we can modify parts of the join/join team buttons? So eg it we define a zombie mod with two teams there won't be a "change team" button, right?

**Comment by necropotame** [comment]
I'm talking about adding new buttons where there is none (right and left side of the screen). For the join/team buttons, we can just add an option to disable it. I need to check if 0.7 vanilla did it depending if some sort of flags.

**Comment by Henningstone** [comment]
Adding new buttons to the esc-screen which can do quite anything the server wants was one of the first features of netgui :)
Modifying or removing the existing ones is not intended in netgui, but it could be added. Ofc some buttons, like disconnect of option should never be modified by the server. But I can do whatever you want :P

**Comment by necropotame** [comment]
Oh, so NetGUI is only accessible from the ESC menu ? In this case, I want the opposite, menu available in game, like a class-chooser. If NetGUI is accessible from everywhere, how did you detect if the ESC menu is displayed ?

**Comment by Henningstone** [comment]
I don't need to detect if the ESC menu is displayed. I only render NetGUI if the ESC-menu is open, like everything else belonging to the teeworlds menus does. But changing that is really not a big deal, and it would be even possible (and also quite useful) if the server could add a flag to the guis if they should be displayed in the menu or as part of the HUD.

**Comment by necropotame** [comment]
Yep, it would be very nice. I think that 3 level of GUI is needed. HUD (no interaction), "dialog mode" (disable the control of the tee, but don't show the ESC menu), and the ESC menu.

**Comment by necropotame** [comment]
What I want is not fully implemented :-p Buttons in the ESC menu have soft shadow oriented differently depending if the button is in the top part or in the bottom part. You can add "NetMsg_ModAPI_GuiEscButton" that add a button (with the shadow and a predefined color) in the left part of the ESC menu. It's more or less a tab system, but visually identified as mod-related. Tell me if you think it's useless ^^

**Comment by Henningstone** [comment]
I assume an additional tab bar at the left, as you mentioned, as completely useless. The mod creators can add buttons onto the whole screen as they like, and shadows could be implemented for sure, though I don't know where and how they are done yet, but will find out about it.
I also get the feeling that you have a certain kind of mod in you mind that should be doable easily using ModAPI. A tabbar would be a pattern that just needs to be filled by the author of this mod.
If any mod author wants a tabbar, they can (and _should_) create it themselves using all the tools the GUI API gives them into their hands. This is what I think about when hearing "ModAPI".

**Comment by necropotame** [comment]
I like some sort of uniformisation. Provide such tab bar will help to keep mods harmonized. I'm working on the statistics system and I think that most of mods must implement a tab with stats for each players (precision for each weapons, number of death, ...). They will also certainly provide an option tab. And I prefer to don't search where is the option tab when I search it in a new mod.

But it's up to you :-)

**Comment by Henningstone** [comment]
Oh well then, I seem to have misunderstood your idea of how ModAPI will be used...
I thought that it is a good thing (and the initial idea) to just give modders some nice tools in their hands, and they create awesome mods with it, completely individual. But if I got it right you really want to give a certain window of possibilities and templates to be used, and in general all ModAPI-mods should be just variations? Hm.

**Comment by necropotame** [comment]
The idea is still to provide a low level API. But some templates could also help moders to start and increase the number if mod. I agree that's not the most important part of ModAPI, and we can leave it to later (or never).
